dashboard/app/api.go

```diff
@@ -1244,15 +1249,33 @@ func needRepro(c context.Context, bug *Bug) bool {
 	return needReproForBug(c, canon)
 }
 
+var syzErrorTitleRe = regexp.MustCompile(`^SYZFAIL:|^SYZFATAL:`)
+
 func needReproForBug(c context.Context, bug *Bug) bool {
-	return bug.ReproLevel < ReproLevelC &&
-		len(bug.Commits) == 0 &&
-		bug.Title != corruptedReportTitle &&
-		bug.Title != suppressedReportTitle &&
-		config.Namespaces[bug.Namespace].NeedRepro(bug) &&
-		(bug.NumRepro < maxReproPerBug ||
-			bug.ReproLevel == ReproLevelNone &&
-				timeSince(c, bug.LastReproTime) > reproRetryPeriod)
+	bestReproLevel := ReproLevelC
+	// For some bugs there's anyway no chance to find a C repro.
+	if syzErrorTitleRe.MatchString(bug.Title) {
+		bestReproLevel = ReproLevelSyz
+	}
+	// We already have a C repro, so do new reproduction only once in a while.
+	if bug.ReproLevel >= bestReproLevel && timeSince(c, bug.LastReproTime) < reproStalePeriod {
+		return false
+	}
+	// There are already fixing commits.
+	if len(bug.Commits) > 0 {
+		return false
+	}
+	if bug.Title == corruptedReportTitle ||
+		bug.Title == suppressedReportTitle {
+		return false
+	}
+	if !config.Namespaces[bug.Namespace].NeedRepro(bug) {
+		return false
+	}
+	// Don't do too many unsuccessful reproductions.
+	return bug.NumRepro < maxReproPerBug ||
```

Comment:
If I am reading this correctly, this will still return false if `bug.NumRepro >= maxReproPerBug` and `timeSince(c, bug.LastReproTime) >= reproStalePeriod`. So we won't get new repros for these bugs.

Reply:
Indeed! Fixed and added a more complicated test.